### DISCUSSION TITLE
New Image BirdNET-Pi_v14.2.img.xz

### DISCUSSION THREAD

**Discussion by mcguirepr89** [mcguirepr89/BirdNET-Pi#353] [Announcements]
THIS IMAGE IS OBSOLETE. INSTALLATION INSTRUCTIONS ON THE MAIN REPO SHOULD BE FOLLOWED INSTEAD.

<details><summary>Obsolete instructions</summary>
<p>
 Hello! I hope this announcement finds you all doing well.

I've put together a new BirdNET-Pi pre-installed image that is ready for **testing**.

The main features of the pre-installed image:
- Contains the auto-hotspot service for easy field access and initial setup
- Comes pre-installed and ready to run
- Configured to write raw data to `/dev/shm` (virtual memory) to minimize SD card read/writes
- `dhcpcd` and `openresolv` have been replaced with a pure `NetworkManager` networking environment
  - Use the command line tools `nmtui` and `nmcli` to manage the system's network connections from the "Web Terminal", SSH, or console.

### Step 1: Download the Image
- [Download the image](https://downloads.pmcgui.xyz/BirdNET-Pi_v14.2.img.xz)

#### Step 1.5 (Highly Recommended): Verify the sha256 checksum
```
04c683cbf8b06b6c54399240a5553c4f295814cc806a4018a1af2c9cf0e65368  BirdNET-Pi_v14.2.img.xz
```

### Step 2: Flash the Image Using the Raspberry Pi Imager
- Open the Raspberry Pi Imager and locate the image downloaded in the step above using the "Choose OS" option as seen in the screenshot below:
  ![image](https://user-images.githubusercontent.com/60325264/172621452-3955d73d-d371-4ade-bef6-d5aab671495b.png)
  ![image](https://user-images.githubusercontent.com/60325264/172621496-ff38ddbc-d89d-4f1b-b2bd-0a7dc7b74558.png)
  ![image](https://user-images.githubusercontent.com/60325264/172621613-4c87e828-0c92-4b1c-b289-77992120b422.png)

- **IMPORTANT!! 
  Remove _any and all_ customized Advanced Settings in Raspberry Pi Imager**
  ![image](https://user-images.githubusercontent.com/60325264/172621704-8d0577aa-a494-4478-bddc-dff86e664092.png)
- Choose the SD card onto which you wish to write the new BirdNET-Pi image
  ![image](https://user-images.githubusercontent.com/60325264/172622323-f1168aee-ab1b-43ec-8b56-a1616258790c.png)
- Write the image to the SD card
  ![image](https://user-images.githubusercontent.com/60325264/172622416-c07c22b2-8e9c-475b-a122-d615c8e8f0fe.png)
  ![image](https://user-images.githubusercontent.com/60325264/172622464-0182e4b4-e860-49fe-be17-e29c1f10611d.png)
  ![image](https://user-images.githubusercontent.com/60325264/172624235-691ffd8a-a9d0-418f-b4d9-5ed6558fbb83.png)

- When you see the dialog below indicating the flashing is complete, you can close the Raspberry Pi Imager and remove the SD.
  ![image](https://user-images.githubusercontent.com/60325264/172624450-0a0ec3d9-29d2-4c16-afaa-dd05af8dc71d.png)

### Step 2: Connect To the BirdNET-Pi HotSpot
- Insert the SD card into the Raspberry Pi and then power it on allowing a few moments to boot up and complete a few first-boot tasks.
- After successfully booting into the image and completing the first-boot tasks, a hotspot to which you can connect called "BirdNET-Pi" will become available. Connect a device to the hotspot to continue setup.

### Step 3: Choosing the Network You Want BirdNET-Pi To Join OR Work Fully Offline
- Once you connect to the "BirdNET-Pi" hotspot, you _should_ automatically be taken to the captive portal where you can either (_**a**_) select your WiFi network to get the BirdNET-Pi connected, or (_**b**_) configure the installation to work fully offline. 
\- **Note:** If you are not automatically taken to the captive portal when you connect to the hotspot, you can navigate there manually by going to http://192.168.42.1.
\- **Note:** Some devices will recognize that the BirdNET-Pi hotspot has no internet access -- be sure your device does not automatically disconnect from the hotspot in an effort to obtain an internet connection. You may need to explicitly instruct your device to stay connected to that hotspot in spite of there being no internet connectivity. On the Apple iPhone, for instance, I must always select to use the network without internet access to stay connected to the hotspot. I only need to do this once.
   - #### (_a_) Connect BirdNET-Pi To a Nearby Network
     ![image](https://user-images.githubusercontent.com/60325264/172647666-eaa275e9-77a0-4389-9663-7eb1b22056bb.png)
     ![image](https://user-images.githubusercontent.com/60325264/172648109-dacae201-7b15-49ab-8228-be998be7a949.png)
     After you enter the WiFi credentials on the captive portal, the BirdNET-Pi will attempt to connect to the network you selected. If it is successful, the "BirdNET-Pi" hotspot will be disconnected and the BirdNET-Pi device will be reachable on your network.
     At that point, join the WiFi network to which you just connected the BirdNET-Pi using any device (laptop, smart phone, PC) and go to http://birdnetpi.local to finish configuring the system.
   - #### (_b_) Configure the Installation To Work Fully Offline
     - Click the blue "Use BirdNET-Pi Offline" link, which will take you to the offline web interface at http://birdnetpi.local:9999
       ![image](https://user-images.githubusercontent.com/60325264/172656121-57e25617-76bf-44ba-8309-43875c42d3d8.png)
     - Proceed to step 4 to configure settings
### Step 4: Configure Settings
**Note:** For the steps below, if you are working offline, the URL in the web browser address bar will actually be "http://birdnetpi.local:9999", otherwise the steps are the same.
- Go to "Tools" > "Settings"
     ![image](https://user-images.githubusercontent.com/60325264/172649786-5756c3ce-ed52-4125-81f9-23bcc177381e.png)
     By default, the username is **birdnet** and the password is empty.
     \- **Note:** If you are not prompted to input the username and password after clicking the "Tools" button and are instead told you cannot edit the settings for the installation, exit the "Captive Portal" and open a regular web browser and go to http://birdnetpi.local:9999 to try again.
     ![image](https://user-images.githubusercontent.com/60325264/172649909-21edb56d-9aa5-408b-98b4-1d757ebb6d3a.png)
     ![image](https://user-images.githubusercontent.com/60325264/172650477-5872c0d5-dd7a-4146-98ee-37201ea347a2.png)
     ![image](https://user-images.githubusercontent.com/60325264/172650675-e93c34f2-dc23-4a6d-9f43-eb4abb2b1319.png)
     Set your latitude and longitude to four decimal places
     ![image](https://user-images.githubusercontent.com/60325264/172650952-36c0912f-24b3-4903-a8f8-d4f59bfcfad9.png)
     **If you are working offline, BE SURE TO SET THE SYSTEM TIME**
     ![image](https://user-images.githubusercontent.com/60325264/172659275-51dcd6e5-b751-40c5-809c-d5b4aa6906b9.png) 

### Step 5: Change system password
Finally, a very important step is changing the default password for the system. To do this, go to "Tools" > "Web Terminal" or via SSH

The username is: `birdie`
The password is: `birdnetpi`

As soon as you are logged in to the shell, type `passwd`

You will be asked for you current password (`birdnetpi`) -- note that you will not see any characters typed as you change the password.
You will then be asked to enter the new password twice.
![image](https://user-images.githubusercontent.com/60325264/172652051-53875eb1-4ef0-4956-a177-cc7c3d468589.png)

That's all!
Happy Birding!
</p></details>

**Comment by ehpersonal38** [comment]
Woah! Good stuff Patrick! I should be able to test this soon

**Reply by mcguirepr89** [reply]
Thanks!

**Comment by nilspupils** [comment]
Will take this with me to the outdoors next week! Thanks Patrick!

**Comment by Joopieh** [comment]
Hi Patrick,

Just did the install as you described above and got it working immediately! So it is safe to say it is monkey proof. 😂. 

I installed it and hooked it up to a powerbank and ran a test with recorded birdsounds from Merlin Bird ID. A great succes!! Tried both English and Dutch and both work. See screenshots below. I hope I have time this weekend to test it in the field. 

Thanks for making this project accessible for people without much knowledge about programming. 

Keep up the good work! 

![image](https://user-images.githubusercontent.com/104306003/172691243-f7b15cfb-768d-4808-b5c6-c5cf3d35a22b.jpeg)

![image](https://user-images.githubusercontent.com/104306003/172691293-b231c576-610b-4831-af20-4821772e353a.png)

**Comment by DD4WH** [comment]
Patrick, thats really awesome! Installed it without any problems.

* I chose the offline version and everything worked perfectly
* then I changed to the WiFi version by going to `http://192.168.42.1/` again and put in my WiFi password
* everything now works in my local WiFi
* and if I want to go offline again, I just switch off my WiFi-Router (the BirdNET-Pi then automatically switches on its hotspot) and can access the BirdNET-Pi again by going to `http://192.168.42.1/` and switch to `offline` 

Perfect! Thanks a lot for this fantastic functionality !

**Reply by DD4WH** [reply]
it seems, in the Settings, if I click `retrieve time automatically`, the system takes a different time zone, it is 6 hours behind
https://github.com/mcguirepr89/BirdNET-Pi/issues/354#issue-1265299562

**Comment by DD4WH** [comment]
Installed the image to a fresh SD card today.
Did an update.
Did the setting of time zone and the other stuff.

Works perfectly! All the fixes seem to be working well now 💪 
Thanks a lot, Ethan & Patrick !!!

**Comment by DCS17** [comment]
Just wanted to say thank you @mcguirepr89  for your work. I've just installed this test version without a hitch. Sounded like a better idea to minimise SD card read/writes!
Working very well on Pi4B 8GB.
What a fascinating new world you have introduced me to!
Unfortunately I'm on 4G in the UK with CGNAT, so I can't publish my page.
Thanks

**Comment by DCS17** [comment]
Hi,
Just done an update as it was highlighted in red on the main screen tools I think it said 53 or 58 - cant remember and it's obviously gone now post update.
Now can't access Tools/Settings with the follwoing error
Fatal error: Uncaught Exception: Unable to open database: unable to open database file in /home/birdie/BirdNET-Pi/scripts/config.php:5 Stack trace: #0 /home/birdie/BirdNET-Pi/scripts/config.php(5): SQLite3->__construct() #1 /home/birdie/BirdNET-Pi/homepage/views.php(134): include('/home/birdie/Bi...') #2 {main} thrown in /home/birdie/BirdNET-Pi/scripts/config.php on line 5

Thanks

**Reply by mcguirepr89** [reply]
I'm very sorry to hear that.

Could you please paste the output of:
```
ls -l ~/BirdNET-Pi/scripts/birds.db
```
and
```
head ~/BirdNET-Pi/scripts/config.php
```

**Reply by ehpersonal38** [reply]
Should the pre installed image be allowed to update from `main` at all? I feel like there's a lot of code conflicts there. Thoughts?

**Reply by DCS17** [reply]
I did hesitate before upgrading because I did wonder. I've learnt the hard way to leave stuff that's working! But too stupid to follow my own lessons !

**Reply by DCS17** [reply]
Thanks for your help
Here's the output.
-rw-rw-r-- 1 birdie birdie 221184 Jun 24 20:19 /home/birdie/BirdNET-Pi/scripts/birds.db

<?php
error_reporting(E_ERROR);
ini_set('display_errors',1);
$db = new SQLite3('./birds.db', SQLITE3_OPEN_CREATE | SQLITE3_OPEN_READWRITE);
function syslog_shell_exec($cmd, $sudo_user = null) {
  if ($sudo_user) {
    $cmd = "sudo -u $sudo_user $cmd";
  }

**Reply by DCS17** [reply]
<?php
error_reporting(E_ERROR);
ini_set('display_errors',1);

$db = new SQLite3('./birds.db', SQLITE3_OPEN_CREATE | SQLITE3_OPEN_READWRITE);

function syslog_shell_exec($cmd, $sudo_user = null) {
  if ($sudo_user) {
    $cmd = "sudo -u $sudo_user $cmd";
  }

**Reply by ehpersonal38** [reply]
Okay I think I've diagnosed the bug. Go ahead and update and let me know if it's working again.

If you can't get to Tools then do a `bash $HOME/scripts/update_birdnet.sh`

**Reply by DCS17** [reply]
Wow! That was quick. Working perfectly again.
Thank you!

**Reply by mcguirepr89** [reply]
@ehpersonal38 is a badass developer

**Reply by ehpersonal38** [reply]
Hahaha, I'd be the shittiest super hero - saving the day by fixing problems I created in the first place

**Comment by DCS17** [comment]


**Reply by DCS17** [reply]
<?php
error_reporting(E_ERROR);
ini_set('display_errors',1);

$db = new SQLite3('./birds.db', SQLITE3_OPEN_CREATE | SQLITE3_OPEN_READWRITE);

function syslog_shell_exec($cmd, $sudo_user = null) {
  if ($sudo_user) {
    $cmd = "sudo -u $sudo_user $cmd";
  }

**Comment by DCS17** [comment]
Wondering about using rclone to backup birds.db and BirdSongs directory to my GoogleDrive using cronjob.
If I install rclone am I going to break something?
I'm a complete novice but do have rclone working well on a PiZeroW used for ADSB monitoring.
Thanks

**Reply by mcguirepr89** [reply]
There should be no conflicts installing `rclone` alongside BirdNET-Pi -- indeed, it looks like a great idea! Let me know how it goes for you --

**Reply by mcguirepr89** [reply]
This may be worth integrating as a backup feature -- I want to look into a web GUI interpretation of the `rclone config` command work flow

:+1: thanks again for the great idea (I know others have recommended a "backup" tool -- this, though, is unambiguously a great choice for the average user -- the more advanced user can opt for `rsync` to use custom storage options if they are unable to get `rclone` to work for their needs)

**Reply by DCS17** [reply]
Looks like I've got rclone working OK. I'm backing up the BirdSongs directory and BirdNET-Pi/scripts/birds.db to my GoogleDrive.
It gives me Charts and recordings as well as detection data.
I couldn't find anything else to back up and tested restoring to an old SD card which seemed to work OK.
Wish I could code well enough to help out incorporating into BirdNetPi. But I can't !

**Reply by ehpersonal38** [reply]
Just got rclone working on a cron myself thanks to your comment. Now my 44MB birds.db is safe!

**Reply by DCS17** [reply]
44MB !!  I've a bit of catching up to do at 250k but gotta start somewhere!

**Reply by mcguirepr89** [reply]
I was able to set it up also ~~ super handy!!! 

How to automate the OAuth Google Client stuff :thinking:

**Reply by ehpersonal38** [reply]
Yeah I have 162k detections at 0.4 min confidence, over only a few months. I think it's getting a little out of hand! 

@mcguirepr89 With some cursory searching I found this:
https://rclone.org/commands/rclone_config_create/
> The flag --non-interactive is for use by applications that wish to configure rclone themeselves, rather than using rclone's text based configuration questions. If this flag is set, and rclone needs to ask the user a question, a JSON blob will be returned with the question in it.

So I guess you could rebuild the interactive `rclone config` flow in HTML and PHP, if you really wanted. 

I'm thinking another way would be just create the config in a single command, and then you can retroactively add the Google drive oauth token to a config file somewhere. (I think [this link](https://rclone.org/remote_setup/#configuring-by-copying-the-config-file) is relevant)

**Reply by mcguirepr89** [reply]
That is helpful -- I was thinking more about [this part](https://rclone.org/drive/#making-your-own-client-id), since I suspect BirdNET-Pi users would max out the rclone API limits

Did you set one of those up? That's what I'd like to automate, but it seems tricky to do that. I used a reverse_proxy in my Caddyfile to redirect the localhost stuff during authorization -- it wasn't super straight forward.

**Reply by ehpersonal38** [reply]
Oh yeah I doubt there's a way to automate that, Google locks all that stuff down to prevent automation. I don't know for sure though.

I ran into the same issue you had during setup. All I had to do to mitigate the localhost stuff was type "n" on this part:
```
Use auto config?
 * Say Y if not sure
 * Say N if you are working on a remote or headless machine
y) Yes (default)
n) No
y/n> n
Please go to the following link: https://accounts.google.com/o/oauth2/auth?access_type=offline&client_id=XXXXXXXXXXXXX...
```

**Reply by mcguirepr89** [reply]
I'm among the unlucky few for whom answering "n" leads me to an error from Google -- I looked into it and  it [seems like it happens to some people and not others](https://stackoverflow.com/questions/71318804/google-oauth-2-0-failing-with-error-400-invalid-request-for-some-client-id-but)

**Comment by dfoster81** [comment]
Bit of an old discussion here, but this image is amazing. I use it on my mobile rig. Installed over Christmas, plugged in to update, and it's just bullet proof in the field. 

Only caveat is you need to set the time every time you start it up. Not sure if this is just a quirk of a non-internet connected raspberry pi, or my particular pi.

I was wondering why there were no detection events for the first part of my walk this morning, only to realize I'd missed the '-' sign in my latitude. Just a constant gotcha in the southern hemisphere :)
